### PR TITLE
adding filter for type of help

### PR DESCRIPTION
**Pull request by cathydeng**
Screen Shot 2022-06-28 at 12 08 45 AM

deets on how the filter should work w the data here: https://app.mural.co/t/publicsafety2218/m/publicsafety2218/1655407064213/4444b12e11e3a1b93756d9eecd808db4f3778de5?sender=ua12e0abf6a5c5b23e7118613

- multiple answers are selectable; if someone chooses multiple we want to surface places that match any of the answers (b/c some combos are unlikely to be served by one provider)
- shortened some of the copy b/c the copy was more for guided search, and was awk in filter ui
- also removed the other/don't know option from the answers in the guided version, b/c the response doesn't make sense in the filter ui since this filter is optional and unpopulated by default
- have not written any tests 🙈 will add to asana

### CODE REVIEW

File: src/components/Search/FiltersControl.tsx [hunk 1]
```diff
@@ -1,7 +1,81 @@
...
+const TypeOfHelpFilter = ({
+  filters,
+  setFilters,
+  t,
+}: {
+  filters: SearchFilters;
+  setFilters: (filters: SearchFilters) => void;
+  t: TFunction;
+}) => {
+  const setTypeOfHelpFilter = (typeOfHelp: TypeOfHelp) => {
```
Comment (jlhogan) [inline src/components/Search/FiltersControl.tsx:50]: add a comment explaining what is happening here please

File: src/components/Search/FiltersControl.tsx [hunk 2]
```diff
@@ -1,7 +1,81 @@
...
+}: {
+  filters: SearchFilters;
+  setFilters: (filters: SearchFilters) => void;
+  t: TFunction;
+}) => {
+  const setDistanceFilter = (miles: number) => {
+    setFilters({ ...filters, miles });
+  };
+
+  const getRadio = (miles: number) => (
```
Comment (jlhogan) [inline src/components/Search/FiltersControl.tsx:20]: alternatively these could all be components but not a blocker for this PR

File: src/util.tsx [hunk 3]
```diff
@@ -17,6 +19,8 @@ export const DEFAULT_RADIUS_MILES = 10;
 
 export const METERS_IN_A_MILE = 1609.34;
 
+export const MILE_DISTANCE_OPTIONS = [10, 25, 50, 100];
```
Comment (jlhogan) [inline src/util.tsx:22]: another argument for pulling each filter out into a component is that these things can live in the same file as the code they pertain to (more helpful for the big chunky filter logic funcs below)

File: src/util.tsx [hunk 4]
```diff
@@ -52,11 +56,51 @@ export const compareDistance = (
...
+export const offersAnyTypesOfHelpNeeded = (
+  careProvider: CareProviderSearchResult,
+  helpNeeded: TypeOfHelp[]
+): boolean => {
+  if (!helpNeeded.length) {
+    // if no help types specified, don't apply any filter
+    return true;
+  } else {
+    // otherwise, check if provider offers ANY of the types of help needed
+    const helpMatches = helpNeeded.map((typeOfHelp) =>
```
Comment (jlhogan) [inline src/util.tsx:91]: as long as you don't need to know how many/ which types of help are offered, you should be able to do the check in the same loop as the transform: `helpNeeded.some((typeOfHelp) => offersTypeOfHelp(careProvider, typeOfHelp))`

**Review by jlhogan — APPROVED** [review]
I think continuing to split things up into small components (e.g. a component for each filter) and keeping related filter logic in those files will keeps things more easily accessible and keep `util` from becoming huge and scary but not a blocker for now!

File: src/components/Search/FiltersControl.tsx (hunk 2 above)
Comment (cathydeng) [reply src/components/Search/FiltersControl.tsx:20]: felt a lil awk having them as separate components b/c rn they can't rly be used alone and dropped somewhere in the absence of the fieldset wrapping them, but will chew on a potential refactor to generalize this more when i implement another set of radios/checkboxes

File: src/util.tsx (hunk 3 above)
Comment (cathydeng) [reply src/util.tsx:22]: i don't think the logic can live purely in the components b/c we'll also need to filter results via guided search - i'm envisioning that the guided search ui will populate url params, and then use the same logic in utils. open to other ideas here tho!

File: src/util.tsx (hunk 4 above)
Comment (cathydeng) [reply src/util.tsx:91]: doh, thx!

**Comment by admoorgit** [comment]
> Screen Shot 2022-06-28 at 12 08 45 AM
Since they've clicked on "Filter" to access this, we don't need to preface each with "Filter by", since that's already established. 

We want to be inclusive of both individual needs AND searching for others — which makes sense, since we're talking to some caregivers this round — and based on the updated content, I'm thinking:

Title of filter (if it's own drop down): "Type of help"

In expanded filter:
“Select what you or someone you care about is looking for help with:"
- Drugs and/or alcohol use
- Court-mandated treatment
- Mental health (persistent feelings of sadness, loneliness, worry....)
- Thoughts about killing themselves, and/or wishing they were dead


NOTE — I have a question in the doc for people, so we'll see if they respond (they haven't been very responsive to me yet, so we'll see...)

**Review by jlhogan — APPROVED** [review]

**Pull request merged**